bin/lint_localized_strings_format

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash -eu
+
+# Attempts to generate the frozen `.strings` file to gets sent to GlotPress
+# after a new release code freeze.
+#
+# If the process fails, annotates the build with the errors.
+
+# The strings generation happens via Fastlane, so we need to install the
+# Ruby gems.
+echo "--- :rubygems: Setting up Gems"
+install_gems
+
+# The strings generation also picks up files from our first party libraries,
+# some of which are installed via CocoaPods
+echo "--- :cocoapods: Setting up Pods"
+install_cocoapods
+
+echo "--- :sleuth_or_spy: Lint Apple Localized Strings Format"
+# A next step improvement is to move the logs management within the release
+# toolkit
+LOGS=logs.txt
+set +e
+set -o pipefail
+bundle exec fastlane generate_strings_file_for_glotpress skip_commit:true | tee $LOGS
```

Comment:
I though of checking whether the lane was available, but there's no out of the box command for that.

If the lane is not there, `fastlane` will return `1`. That will make the lane fail and we'll print the error via the code below.

```
$ bundle exec fastlane lane_name_thats_not_available

[...]
[!] Could not find lane 'ios lane_name_thats_not_available'. Available lanes: ios build_for_testing, ios build_jetpack_for_testing, ios test_without_building, ...
```

The build annotation will likely end up being empty. 

This should be an edge case that would occur only when setting up a new project or in the unlikely occurrence that we'll change the `generate_strings_file_for_glotpress` lane name. Having refined error handling for this occurrence seems hitting diminishing returns.

Reply:
I agree.

Though one easy thing we could consider doing, is to provide this as a default-but-overridable parameter, [like how we do it for commands like `build_and_test_pod`](https://github.com/Automattic/bash-cache-buildkite-plugin/blob/trunk/bin/build_and_test_pod#L27) that uses `test` as the lane name by default but allows to override it by providing it as `$1` (well, `$@` in this case)

Reply:
```suggestion
bundle exec fastlane "${@:-generate_strings_file_for_glotpress}" skip_commit:true | tee $LOGS
```

Reply:
@AliSoftware great suggestion, thanks!

I was curious about the `$@` usage, tried it locally, and run in some issues. 

If we use `"${@...}"`, that is wrap the expansion in double quote to avoid re-splitting elements like ShellCheck suggests, and also pass more than one argument to the script invocation, the double quotes will result in Fastlane not finding our lane.

I noticed this when I tried to be smart and move the `skip_commit:true` in the fallback:

```
bundle exec fastlane "${@:-generate_strings_file_for_glotpress skip_commit:true}"
```

<img width="1260" alt="image" src="https://user-images.githubusercontent.com/1218433/202834842-e7b6d816-3bc7-42be-89ba-f40565c37781.png">

The approach works fine without the double quotes. However, ShellCheck raises a warning, `SC2068`.


```
bundle exec fastlane ${@:-generate_strings_file_for_glotpress skip_commit:true}
```

<img width="1120" alt="image" src="https://user-images.githubusercontent.com/1218433/202834922-4b77dfdb-4397-4d62-b4d0-53fa491720c7.png">

At this point, I'd like to use the following, what do you think?

```suggestion
# We cannot wrap `${@:-...}` in double quotes because of Fastlane.
# If the caller passes more than one parameter, Fastlane will consider the resulting string with spaces as the name of the lane to run failing the script.
# That's the same if our fallback, which constains more than one word, takes place.
#
# shellcheck disable=2068
bundle exec fastlane ${@:-generate_strings_file_for_glotpress skip_commit:true} | tee $LOGS
```

Reply:
`"${@:-default}"` will wrap each individual argument from $@ in quotes, which is exactly what we want: if we use that and then  pass `lint_localized_strings_format a b` then `"${@:-default}"` will be substituted with `"a" "b"` (as opposed to `"a b"` like one might think at first). So I still think that's what we need to use.

But when using this syntax, `default` needs to be a single default argument. That means that if you use `"${@:-d1 d2}"` then pass no argument, it will be substituted with the single default argument `"d1 d2"` this time, not `"d1" "d2"`. This is why when you try to include `skip_commit:true` inside it it will not find the lane as it will try to find a lane whose name is `"ios_generate_strings_file_for_glotpress skip_commit:true"` which doesn't exists.

---

I see two solutions for this:

1. Use my initial suggestion. That will always include `skip_commit:true` as part of the parameters passed to the lane. But I don't think that's a problem because (1) I don't see a use case why we'd want `skip_commit:false` instead and (2) if the repo we're running this in uses a different name for that option (that we'd then pass as part of the `$@` parameters for this case), including an unused option to the lane call will not have any side effects and will just be ignored (that lane from that repo will be unlikely to use `options[:skip_commit]` for anything other than deciding if we need to skip the commit step or not)

2. Be more clever and inject `skip_commit:true` into the `$@` parameters list (via `unshift` or something like that?) only `if [[ $# -eq 0 ]]`, before the line that then uses `"$@"`. In other words, don't use the `:-` fallback mechanism in the `"$@"` syntax but instead modify the content of `$@` before if no parameter is passed.

I don't really think solution 2 is worth it though, and I think solution 1 is totally enough for our use case, allowing to overwrite the lane name if needed and pass _additional_ arguments to it in addition to `skip_commit:true` that will always be passed.

Reply:
> `"${@:-default}"` will wrap each individual argument from `$@` in quotes, which is exactly what we want

I should have tested that scenario better.

> Be more clever and inject `skip_commit:true` into... only `if [[ $# -eq 0 ]]`...

This suggestion made me think of a maybe "less clever" but equally effective alternative:

```sh
if [[ $# -eq 0 ]]; then
  bundle exec fastlane update_appstore_strings skip_commit:true
else
  bundle exec fastlane "$@"
fi
```

The reason I'm keen to have either `$@` or the lane name including all its parameters is because I think if we provide a way for the user to override the default, we should go all in with it.

You made a compelling case for why having `skip_commit:true` as an argument in both scenarios is not an issue and I think it would be a reasonable compromise.

However, I can see myself reading the code in the future and asking why `skip_commit:true` I outside the `${@:-...}` variable substitution. We could easily link to this conversation or leave a comment that summarizes it, but the code above, unless I missed something in the way it will behave, speak for itself.

Reply:
I like that compromise suggestion of yours 👍  Let's go with it!